tests/object.cpp

```diff
@@ -18,23 +18,99 @@
 
+namespace {
+using AnyDict = std::map<std::string, util::Any>;
+using AnyVec = std::vector<util::Any>;
+using Defaults = std::map<std::string, AnyDict>;
+}
+
+namespace realm {
+template<>
+class NativeAccessor<util::Any, Defaults*> {
+public:
+    static bool dict_has_value_for_key(Defaults*, util::Any dict, const std::string &prop_name)
+    {
+        return any_cast<AnyDict>(dict).count(prop_name) != 0;
+    }
+
+    static util::Any dict_value_for_key(Defaults*, util::Any dict, const std::string &prop_name)
+    {
+        return any_cast<AnyDict>(dict).at(prop_name);
+    }
+
+    static size_t list_size(Defaults*, util::Any& v) { return any_cast<AnyVec>(v).size(); }
+    static util::Any list_value_at_index(Defaults*, util::Any& v, size_t index)
+    {
+        return any_cast<AnyVec>(v)[index];
+    }
+
+    static bool has_default_value_for_property(Defaults* def, Realm*, ObjectSchema const& object, std::string const& prop)
+    {
+        auto it = def->find(object.name);
+        if (it != def->end())
+            return it->second.count(prop);
+        return false;
+    }
+
+    static util::Any default_value_for_property(Defaults* def, Realm*, ObjectSchema const& object, std::string const& prop)
+    {
+        return def->at(object.name).at(prop);
+    }
+
+    static Timestamp to_timestamp(Defaults*, util::Any& v) { return any_cast<Timestamp>(v); }
+    static bool to_bool(Defaults*, util::Any& v) { return any_cast<bool>(v); }
+    static double to_double(Defaults*, util::Any& v) { return any_cast<double>(v); }
+    static float to_float(Defaults*, util::Any& v) { return any_cast<float>(v); }
+    static long long to_long(Defaults*, util::Any& v) { return any_cast<long long>(v); }
+    static std::string to_binary(Defaults*, util::Any& v) { return any_cast<std::string>(v); }
+    static std::string to_string(Defaults*, util::Any& v) { return any_cast<std::string>(v); }
+    static Mixed to_mixed(Defaults*, util::Any&) { throw std::logic_error("'Any' type is unsupported"); }
+
+    static util::Any from_binary(Defaults*, BinaryData v) { return std::string(v); }
+    static util::Any from_bool(Defaults*, bool v) { return v; }
+    static util::Any from_double(Defaults*, double v) { return v; }
+    static util::Any from_float(Defaults*, float v) { return v; }
+    static util::Any from_long(Defaults*, long long v) { return v; }
+    static util::Any from_string(Defaults*, StringData v) { return std::string(v); }
+    static util::Any from_timestamp(Defaults*, Timestamp v) { return v; }
+    static util::Any from_list(Defaults*, List v) { return v; }
+    static util::Any from_results(Defaults*, Results v) { return v; }
+    static util::Any from_object(Defaults*, Object v) { return v; }
+
+    static bool is_null(Defaults*, util::Any& v) { return !v.has_value(); }
+    static util::Any null_value(Defaults*) { return {}; }
+
+    static size_t to_existing_object_index(Defaults*, SharedRealm, util::Any &) { return 0; }
```

Comment:
It'd be clearer to have this be something like `REALM_ASSERT(false && "Not implemented");`, so it's obvious to anyone adding new tests that this needs to be fleshed out before it can be tested.